### PR TITLE
AP-1681 Events not stored in context so no conversions recorded

### PR DESCRIPTION
**Pull request by robertsevernsentient**

### CODE REVIEW

File: EvolvSwiftSDK/EvolvSwiftSDKTests/Common-Tests/EvolvClientTests.swift [hunk 1]
```diff
@@ -130,7 +131,8 @@ class EvolvClientTests: XCTestCase {
         
         let actualSubmittedEvents = evolvAPI.submittedEvents as! [EvolvContamination]
         let expectedSubmittedEvents: [EvolvContamination] = [
-            .init(cid: "5fa0fd38aae6:47d857cd5e", uid: "C51EEAFC-724D-47F7-B99A-F3494357F164", eid: "ff01d1516c", timeStamp: actualSubmittedEvents[0].timeStamp, contaminationReason: contaminationReason)
+            .init(cid: "5fa0fd38aae6:47d857cd5e", uid: "C51EEAFC-724D-47F7-B99A-F3494357F164", eid: "ff01d1516c", timeStamp: actualSubmittedEvents[0].timeStamp, contaminationReason: contaminationReason),
```
Comment (mattstrom) [inline EvolvSwiftSDK/EvolvSwiftSDKTests/Common-Tests/EvolvClientTests.swift:134]: What does this syntax do??
```swift
[.init()]
```

File: EvolvSwiftSDK/EvolvSwiftSDKTests/Common-Tests/EvolvClientTests.swift [hunk 2]
```diff
@@ -145,8 +147,11 @@ class EvolvClientTests: XCTestCase {
         client.contaminate(details: contaminationReason, allExperiments: false)
 
         let actualSubmittedEvents = evolvAPI.submittedEvents as! [EvolvContamination]
-        let expectedSubmittedEvents = [EvolvContamination]()
+        let expectedSubmittedEvents: [EvolvContamination] = [
+            .init(cid: "5fa0fd38aae6:1234567890", uid: "C51EEAFC-724D-47F7-B99A-F3494357F164", eid: "1234567890", timeStamp: actualSubmittedEvents[0].timeStamp, contaminationReason: contaminationReason)
+        ]
 
+        // TODO it's more acurate for this to fire no contaminations. As, whilst the experiment audience was matched, there were no active keys
```
Comment (mattstrom) [inline EvolvSwiftSDK/EvolvSwiftSDKTests/Common-Tests/EvolvClientTests.swift:154]:
```suggestion
        // TODO it's more accurate for this to fire no contaminations. As, whilst the experiment audience was matched, there were no active keys
```

File: EvolvSwiftSDK/EvolvSwiftSDK/Sources/Core/EvolvClientImpl.swift [hunk 3]
```diff
@@ -248,8 +248,9 @@ final class EvolvClientImpl: EvolvClient {
     public func emit<T: Encodable>(eventType: String, metadata: T?, flush: Bool) {
         let event = EvolvCustomEventForSubmission(type: eventType, uid: options.participantID, metadata: metadata)
         
+        let oldEvents = evolvStore.evolvContext.events;
         evolvStore.evolvContext.events.append(.init(type: eventType, timestamp: Date()))
-        
+        self.evolvStore.evolvContext.contextChanged(key: "events", value: evolvStore.evolvContext.events, before: oldEvents)
```
Comment (mattstrom) [inline EvolvSwiftSDK/EvolvSwiftSDK/Sources/Core/EvolvClientImpl.swift:253]: Is `evolvStore.evolvContext.events` passed by value or by reference? Or rather, is `evolvStore.evolvContext.events` treated as immutable once `contextChanged` receives it?

Comment (robertsevernsentient) [reply EvolvSwiftSDK/EvolvSwiftSDK/Sources/Core/EvolvClientImpl.swift:253]: From this it seemed that shallow copies here are by value
https://stackoverflow.com/questions/27812433/how-do-i-make-a-exact-duplicate-copy-of-an-array

**Review by mattstrom — APPROVED** [review]

**Pull request merged**